app/controllers/mechanics_controller.rb

```diff
@@ -0,0 +1,13 @@
+class MechanicsController < ApplicationController
+  def show
+    @mechanic = Mechanic.find(params[:id])
+  end
+
+  def update
+    # require 'pry'; binding.pry
+    @mechanic = Mechanic.find(params[:id])
+    @ride = Ride.find(params[:mechanic][:ride_id])
+    @mechanic.rides << @ride
+    redirect_to mechanic_path(@mechanic)
+  end
```

Comment:
When looking at US2, here is what we're being asked to do: `"When I go to a mechanic's show page, I see a form to ADD a ride to their workload with an ID of an existing ride"`.

I can see that you're trying to do that using an `update` but because we are CREATING a new relationship we want this to in the `create` action.

The only other thing is where this operation needs to take place.  Since `a mechanic can HAVE MANY rides` and `a ride can HAVE MANY mechanics`, when adding an additional ride to a mechanic, this operation should happen in our joins controller, `MechanicRidesController`.

Here is what the code could look like:

```ruby
class MechanicRidesController < ApplicationController
  def create
    @mechanic = Mechanic.find(params[:id])
    @ride = Ride.find(params[:mechanic][:ride_id])
    @mechanic.rides << @ride
    redirect_to mechanic_path(@mechanic)
  end
end
```

So long story short, the implementation is correction, the location of both the action and controller aren't.